### ISSUE TITLE
technical.py: RSI trata series planas como RSI=100 en lugar de RSI=50

### ISSUE THREAD

**Issue by lucasabu1988** [lucasabu1988/NuevoProyecto#28]
## Bug
`calculate_rsi()` puede devolver `RSI=100` en series sin cambios (flat), porque la pérdida media se convierte en 0 y se rellena con `100.0` por defecto.

## Archivo
`src/signals/technical.py`

## Detalle
```python
gain = (deltas.where(deltas > 0, 0)).rolling(window=period).mean()
loss = (-deltas.where(deltas < 0, 0)).rolling(window=period).mean()

rs = gain / loss.replace(0, np.nan)
rsi = 100 - (100 / (1 + rs))
rsi = rsi.fillna(100.0)  # loss=0 means all gains, RSI=100
```

En una serie plana, `gain = 0` y `loss = 0`, entonces `rs` es NaN y se rellena como `100.0`, aunque no haya habido ni ganancias ni pérdidas.

## Impacto
- Distorsión del RSI hacia sobrecompra en activos sin movimiento.
- Señales erróneas de `RSI_OVERBOUGHT` en activos flat o con datos corruptos.

## Comportamiento esperado
- Series sin cambios → RSI ≈ 50 (neutral)
- Series con solo ganancias → RSI = 100
- Series con solo pérdidas → RSI = 0

## Fix sugerido
Distinguir los casos antes del `fillna`:

```python
# En lugar de fillna(100.0) incondicional:
# gain=0 y loss=0 → neutral (50), no sobrecompra (100)
rs = gain / loss.replace(0, np.nan)
rsi = 100 - (100 / (1 + rs))

# Solo rellenar con 100 cuando haya ganancias pero cero pérdidas
mask_all_gain = (gain > 0) & (loss == 0)
mask_flat = (gain == 0) & (loss == 0)
rsi = rsi.fillna(100.0)
rsi[mask_flat] = 50.0
```

## Criterio de aceptación
- Series planas producen RSI ≈ 50.
- El caso "solo ganancias" sigue dando RSI=100.
- El caso "solo pérdidas" da RSI=0.
